x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/wired_stream.ts

```diff
@@ -0,0 +1,610 @@
+              name: parentId,
+              ingest: {
+                ...parentStream.definition.ingest,
+                wired: {
+                  ...parentStream.definition.ingest.wired,
+                  routing: [
+                    ...currentParentRouting,
+                    {
+                      destination: this._definition.name,
+                      if: { never: {} },
+                    },
+                  ],
+                },
+              },
+            },
+          });
+        }
+      }
+    }
+
+    return { cascadingChanges, changeStatus: 'upserted' };
+  }
+
+  protected async doHandleDeleteChange(
+    target: string,
+    desiredState: State,
+    startingState: State
+  ): Promise<{ cascadingChanges: StreamChange[]; changeStatus: StreamChangeStatus }> {
+    if (target !== this._definition.name) {
+      return { cascadingChanges: [], changeStatus: this.changeStatus };
+    }
+
+    const cascadingChanges: StreamChange[] = [];
+    const parentId = getParentId(this._definition.name);
+    if (parentId) {
+      const parentStream = desiredState.get(parentId) as WiredStream;
+      // if the parent hasn't changed already, ensure it has the correct routing
+      // if it has changed, the routing will be updated in the parent's upsert
+      if (!parentStream.hasChanged()) {
+        const currentParentRouting = parentStream.definition.ingest.wired.routing;
+        const hasChild = currentParentRouting.some(
+          (routing) => routing.destination === this._definition.name
+        );
+        if (hasChild) {
+          cascadingChanges.push({
+            type: 'upsert',
+            definition: {
+              name: parentId,
+              ingest: {
+                ...parentStream.definition.ingest,
+                wired: {
+                  ...parentStream.definition.ingest.wired,
+                  routing: parentStream.definition.ingest.wired.routing.filter(
+                    (routing) => routing.destination !== this._definition.name
+                  ),
+                },
+              },
+            },
+          });
+        }
+      }
+    }
+
+    cascadingChanges.push(
+      ...this._definition.ingest.wired.routing.map((routing) => ({
+        name: routing.destination,
+        type: 'delete' as const,
+      }))
+    );
+
+    return { cascadingChanges, changeStatus: 'deleted' };
+  }
+
+  protected async doValidateUpsertion(
+    desiredState: State,
+    startingState: State
+  ): Promise<ValidationResult> {
+    if (!hasSupportedStreamsRoot(this._definition.name)) {
+      return {
+        isValid: false,
+        errors: ['Cannot create wired stream due to unsupported root stream'],
+      };
+    }
+
+    const existsInStartingState = startingState.has(this._definition.name);
+
+    if (!existsInStartingState) {
+      // TODO in this check, make sure the existing data stream is not a stream-created one (if it is, state might be out of sync, but we can fix it)
```

Comment:
should we still do this? I think I put this here, it's about if the data stream exists but we created it, then it's probably fine to "grandfather" it in. But we also can not do this for now

Reply:
I'll skip it for now